neuralop/layers/mwno_block.py

```diff
@@ -0,0 +1,701 @@
+        P'_k(x) = (2k+1)P_{k-1}(x) + (2(k-2)+1)P_{k-3}(x) + ... 
+        
+        Parameters
+        ----------
+        k : int
+            Order of the Legendre polynomial
+        x : array_like
+            Points at which to evaluate the derivative
+            
+        Returns
+        -------
+        out : array_like
+            Derivative values at the given points
+        """
+
+        def _legendre(k, x):
+            return (2*k+1) * eval_legendre(k, x)
+        out = 0
+        for i in np.arange(k-1,-1,-2):
+            out += _legendre(i, x)
+        return out
+
+    @staticmethod
+    def phi_(phi_c, x, lb=0, ub=1):
+        """
+        Evaluate a polynomial basis function with compact support.
+        
+        This function evaluates a polynomial defined by coefficients phi_c,
+        but sets the value to zero outside the interval [lb, ub].
+        
+        Parameters
+        ----------
+        phi_c : array_like
+            Polynomial coefficients
+        x : array_like
+            Points at which to evaluate the polynomial
+        lb : float, optional
+            Lower bound of the support interval, default 0
+        ub : float, optional
+            Upper bound of the support interval, default 1
+            
+        Returns
+        -------
+        array_like
+            Polynomial values, zero outside [lb, ub]
+        """
+        mask = np.logical_or(x<lb, x>ub) * 1.0
+        return np.polynomial.polynomial.Polynomial(phi_c)(x) * (1-mask)
+
+    @classmethod
+    def get_phi_psi(cls, k, base):
+        """
+        Generate scaling functions (phi) and wavelet functions (psi) for the given basis.
+        
+        This method constructs orthogonal scaling and wavelet functions using either
+        Legendre or Chebyshev polynomials as the base. The construction follows a
+        Gram-Schmidt-like orthogonalization procedure.
+        
+        Parameters
+        ----------
+        k : int
+            Number of basis functions (determines the approximation order)
+        base : str
+            Type of polynomial basis ('legendre' or 'chebyshev')
+            
+        Returns
+        -------
+        phi : list of functions
+            List of k scaling functions
+        psi1 : list of functions
+            List of k wavelet functions for the first half of the interval
+        psi2 : list of functions
+            List of k wavelet functions for the second half of the interval
+            
+        Notes
+        -----
+        For Legendre basis:
+        - Scaling functions are normalized Legendre polynomials on [0,1]
+        - Wavelets are constructed via orthogonalization against scaling functions
+        
+        For Chebyshev basis:
+        - Scaling functions are normalized Chebyshev polynomials
+        - Wavelets have compact support on [0,0.5] and [0.5,1] respectively
+        """
+        x = Symbol('x')
```

Comment:
Do we need to go through Sympy for this? Couldn't we directly return the discretized evaluations, instead of generating a symbolic version and then evaluating?